plugins/woocommerce/includes/admin/class-wc-admin.php

```diff
@@ -289,6 +291,57 @@ public function preview_email_editor_dummy_content() {
 		exit;
 	}
 
+	/**
+	 * Log a warning if checkout/shipping/billing field filters are used without locale-aware filters.
+	 */
+	public function wc_log_missing_locale_filters() {
+		$uses_checkout_fields = has_filter( 'woocommerce_checkout_fields' );
+		$uses_shipping_fields = has_filter( 'woocommerce_shipping_fields' );
+		$uses_billing_fields  = has_filter( 'woocommerce_billing_fields' );
+
+		$uses_field_filters = $uses_checkout_fields || $uses_shipping_fields || $uses_billing_fields;
+
+		$uses_default_address_fields     = has_filter( 'woocommerce_default_address_fields' );
+		$uses_get_country_locale         = has_filter( 'woocommerce_get_country_locale' );
+		$uses_get_country_locale_default = has_filter( 'woocommerce_get_country_locale_default' );
+
+		$uses_locale_filters = $uses_default_address_fields || $uses_get_country_locale || $uses_get_country_locale_default;
+
+		if ( $uses_field_filters && ! $uses_locale_filters && function_exists( 'wc_get_logger' ) ) {
+			wc_get_logger()->warning(
+				'Detected use of checkout/billing/shipping field filters without corresponding locale filters. This may result in incomplete or misconfigured address fields at checkout. Encourage plugin developers to also use woocommerce_default_address_fields or woocommerce_get_country_locale / woocommerce_get_country_locale_default filters.',
```

Comment:
I suggested that the other message is updated to be translatable, but noting that this one is fine to keep in English since it only goes into the logs.